src/uct/ib/mlx5/dv/ib_mlx5dv_md.c

```diff
@@ -1206,29 +1251,54 @@ static ucs_status_t uct_ib_mlx5dv_check_dc(uct_ib_device_t *dev)
     attr.ah_attr.grh.traffic_class = 0;
     attr.ah_attr.grh.sgid_index    = 0;
 
-    ret = ibv_modify_qp(qp, &attr, IBV_QP_STATE |
-                                   IBV_QP_MIN_RNR_TIMER |
-                                   IBV_QP_AV |
-                                   IBV_QP_PATH_MTU);
+    ret = ibv_modify_qp(dct_qp, &attr, IBV_QP_STATE |
+                                       IBV_QP_MIN_RNR_TIMER |
+                                       IBV_QP_AV |
+                                       IBV_QP_PATH_MTU);
 
     if (ret != 0) {
         ucs_debug("%s: failed to ibv_modify_qp(DCT, RTR): %m",
                   uct_ib_device_name(dev));
         status = UCS_OK;
-        goto out_destroy_qp;
+        goto out_destroy_dct_qp;
+    }
+
+    qp_attr.send_cq             = cq;
+    qp_attr.recv_cq             = cq;
+    qp_attr.srq                 = srq;
+    qp_attr.qp_type             = IBV_QPT_DRIVER;
+    qp_attr.sq_sig_all          = 0;
+    qp_attr.comp_mask           = IBV_QP_INIT_ATTR_PD;
+    qp_attr.pd                  = pd;
+    qp_attr.cap.max_send_wr     = 128;
+    qp_attr.cap.max_recv_wr     = 0;
+    uct_ib_mlx5_dci_qp_update_attr(&qp_attr);
+
+    uct_ib_mlx5dv_dc_qp_init_attr(&dv_attr, MLX5DV_DCTYPE_DCI);
+    dci_qp = UCS_PROFILE_CALL_ALWAYS(mlx5dv_create_qp, ctx, &qp_attr,
+                                     &dv_attr);
+    if (dci_qp == NULL) {
+        ucs_debug("%s: mlx5dv_create_qp(DCT) failed: %m",
+                  uct_ib_device_name(dev));
+        status = UCS_OK;
+        goto out_destroy_dct_qp;
+    }
+
+    status = uct_ib_mlx5dv_tx_wqe_ratio(dci_qp, &qp_attr,
+                                        &md->tx_wqe_ratio.dc);
+    if (status != UCS_OK) {
```

Comment:
this code is almost the same as `uct_ib_mlx5dv_check_dc`, plz unify

Reply:
but it is `uct_ib_mlx5dv_check_dc`

Reply:
unified `qp_init_attr` for `uct_ib_mlx5dv_check_dc` and `uct_ib_mlx5dv_check_ud`